### PR TITLE
(dsl): Support 'max' aggregation

### PR DESCRIPTION
**Pull request by dbulaja98**
Resolves `maxAgg` in #118

### CODE REVIEW

File: modules/library/src/main/scala/zio/elasticsearch/aggregation/Aggregations.scala [hunk 1]
```diff
@@ -31,6 +31,22 @@ sealed trait ElasticAggregation { self =>
 
 sealed trait SingleElasticAggregation extends ElasticAggregation
 
+sealed trait MaxAggregation extends SingleElasticAggregation with HasMissing[MaxAggregation] with WithAgg
+
+final case class Max(name: String, field: String, missing: Option[Double]) extends MaxAggregation { self =>
```
Comment (drmarjanovic) [inline modules/library/src/main/scala/zio/elasticsearch/aggregation/Aggregations.scala:36]: Should this be private?

File: modules/library/src/main/scala/zio/elasticsearch/aggregation/options/HasMissing.scala [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package zio.elasticsearch.aggregation.options
+
+private[elasticsearch] trait HasMissing[A <: HasMissing[A]] {
+
+  /**
+   * Sets the `missing` parameter for the [[zio.elasticsearch.aggregation.ElasticAggregation]]. The`Missing` parameter
```
Comment (drmarjanovic) [inline modules/library/src/main/scala/zio/elasticsearch/aggregation/options/HasMissing.scala:22]:
```suggestion
   * Sets the `missing` parameter for the [[zio.elasticsearch.aggregation.ElasticAggregation]]. The `missing` parameter
```

**Review by drmarjanovic — CHANGES_REQUESTED** [review]

File: modules/library/src/it/scala/zio/elasticsearch/HttpExecutorSpec.scala [hunk 3]
```diff
@@ -106,7 +126,7 @@ object HttpExecutorSpec extends IntegrationSpec {
             Executor.execute(ElasticRequest.createIndex(firstSearchIndex)),
             Executor.execute(ElasticRequest.deleteIndex(firstSearchIndex)).orDie
           ),
-          test("aggregate using terms aggregation with nested terms aggregation") {
+          test("aggregate using terms aggregation with nested max aggregation") {
```
Comment (arnoldlacko) [inline modules/library/src/it/scala/zio/elasticsearch/HttpExecutorSpec.scala:129]: Is this meant to be replacing the `terms` test, do we not want to keep both?

Comment (arnoldlacko) [reply modules/library/src/it/scala/zio/elasticsearch/HttpExecutorSpec.scala:129]: I see the point, it's nesting aggregations of different type.

**Review by drmarjanovic — APPROVED** [review]

**Pull request merged**